### PR TITLE
Docker image build

### PR DESCRIPTION
**Pull request by juliushaertl**
Github action to build docker images for amd64/arm64 and push them to ghcr.io

We'll have a daily tag following main and latest for tagged releases

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,9 +1,9 @@
 # SPDX-FileCopyrightText: 2024 Nextcloud GmbH and Nextcloud contributors
 # SPDX-License-Identifier: AGPL-3.0-or-later
 
-ARG NODE_VERSION=20.0.0
+ARG NODE_VERSION=20
 
-FROM node:${NODE_VERSION}-alpine
```
Comment (szaimen) [inline Dockerfile:6]: so not alpine version anymore?

File: .github/workflows/docker.yml [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+        platform:
+          - linux/amd64
+          - linux/arm64
+
+    steps:
+      - name: Set vars
+        id: vars
+        run: |
+          echo "version_channel=${{ github.event_name == 'release' && 'release' || 'daily' }}" >> $GITHUB_OUTPUT
+          echo "version=${{ github.sha }}" >> $GITHUB_OUTPUT
```
Comment (szaimen) [inline .github/workflows/docker.yml:34]: so not version tag like e.g. v1.0.0 or comparable?

File: Dockerfile (hunk 1 above)
Comment (juliushaertl) [reply Dockerfile:6]: I had some trouble there with the arm build, but can see if I get it working somehow in a follow up

File: .github/workflows/docker.yml (hunk 2 above)
Comment (juliushaertl) [reply .github/workflows/docker.yml:34]: Ah right we should also tag those 👍

File: Dockerfile (hunk 1 above)
Comment (szaimen) [reply Dockerfile:6]: Follow-up with Alpine would be cool as packages are generally more up-to-date

Comment (juliushaertl) [reply Dockerfile:6]: Pushed, seemed to have worked without issues now. Not sure what the problem was before

**Comment by szaimen** [comment]
@juliushaertl I looked at https://github.com/nextcloud/whiteboard/pkgs/container/whiteboard/versions?filters%5Bversion_type%5D=tagged, and it looks like no image with e.g. the tag 0.3.0-beta.1 https://github.com/nextcloud/whiteboard/blob/195a6db4b67a3a3d179d95f1cd975559ffc7ba66/appinfo/info.xml#L16 is available... Would be useful in case for AIO so that we can pin the version without having to look for changes upstream...

**Comment by szaimen** [comment]
Or will this be only created after the next release?

**Comment by juliushaertl** [comment]
We didn't plan to tag releases right away, but more towards the nearing final for 30

**Comment by szaimen** [comment]
> We didn't plan to tag releases right away, but more towards the nearing final for 30

all right 👍

**Pull request merged**